x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/external/openai/OpenAiResponseHandlerTests.java

```diff
@@ -31,33 +30,47 @@ public void testCheckForFailureStatusCode() {
 
         var httpResult = new HttpResult(httpResponse, new byte[] {});
 
+        var handler = new OpenAiResponseHandler("", result -> null);
+
         // 200 ok
-        OpenAiResponseHandler.checkForFailureStatusCode(httpRequest, httpResult);
+        when(statusLine.getStatusCode()).thenReturn(200);
```

Comment:
Moving the mock here because the refactor I did caused additional calls to `getStatusCode` and I figured it looks worse to have `when(statusLine.getStatusCode()).thenReturn(200).thenReturn(200).thenReturn(503).thenReturn(503)...`